src/core/mac/mac.cpp

```diff
@@ -1595,6 +1595,26 @@ void Mac::HandleReceiveTimer(void)
     }
 }
 
+otError Mac::ProcessFrameSequence(Frame &aFrame, const Address &aSrcAddr)
+{
+    otError   error        = OT_ERROR_NONE;
+    Neighbor *linkNeighbor = GetNetif().GetMle().GetLinkNeighbor(aSrcAddr);
+
+    VerifyOrExit(linkNeighbor != NULL);
+
+    if (aFrame.GetSequence() + 1 <= linkNeighbor->GetLinkFrameSequence())
```

Comment:
How to handle `uint8_t` wrap around and possibly missing frames ?

For example, if `GetLinkFrameSequence()` is at `0xff`, and we miss a frame (miss receiving sequence number `0xff` itself), then we reject the next frame with seq number `0` and all the next ones (which are all technically ahead/after sequence number `0xff`)